app.py

```diff
@@ -1,37 +1,47 @@
-from flask import Flask, request, render_template_string, redirect, url_for
+from flask import Flask, request, render_template, redirect, url_for
 import json
 import os 
 app= Flask(__name__)
-DATA_FILE = "mooduri.json"
+DATA_FILE = "moods.json"
 
-if os.path.exists(DATA_FILE):
-    with open(DATA_FILE, "r") as f:
-        moods = json.load(f)
-else:
-    moods = []
+moods = []
+
+def load_moods():
+    if os.path.exists(DATA_FILE):
+        with open(DATA_FILE, "r") as f:
+            try:
+                return json.load(f)
+            except json.JSONDecodeError:
+                return []
+    return[]
+
+def save_moods():
+    with open(DATA_FILE,"w") as f:
+        json.dump(moods, f)
 
 @app.route("/", methods=["GET","POST"])
 def home():
     global moods
```

Comment:
Nu se modifica `moods` aici, nu ai nevoie de `global`

`global` se foloseste atunci cand vrei sa modifici variabile globale

e.g.

```py
x = [1, 2, 3]

def f():
     # Nu ai nevoie de global
     x.append(5)
     
def g():
    # Ai nevoie de global, se modifica referinta catre variabila
    global x
    x = [2, 3, 4]
```

Daca nu se folosea `global x` in `g()`, s-ar fi creeat o variabila *locala* numita `x`